File: app/src/main/java/cc/smartcash/wallet/Adapters/DashboardWalletListAdapter.java
package cc.smartcash.wallet.Adapters;

import android.annotation.SuppressLint;
import android.content.Context;
import android.graphics.Color;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import cc.smartcash.wallet.Models.Coin;
import cc.smartcash.wallet.Models.Wallet;
import cc.smartcash.wallet.R;
import cc.smartcash.wallet.Utils.SmartCashApplication;
import cc.smartcash.wallet.ViewHolders.DashboardWalletListViewHolder;

public class DashboardWalletListAdapter extends RecyclerView.Adapter<DashboardWalletListViewHolder> {

    private Context context;
    private ArrayList<Wallet> wallets;
    private Integer count;

    public DashboardWalletListAdapter(Context context, ArrayList<Wallet> wallets) {
        this.context = context;
        this.wallets = wallets;
        this.count = 0;
    }

    public void setItems(ArrayList<Wallet> wallets) {
        this.wallets = wallets;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public DashboardWalletListViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(context).inflate(R.layout.card_wallet_item, viewGroup, false);
        return new DashboardWalletListViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DashboardWalletListViewHolder walletViewHolder, int i) {

        SmartCashApplication app = new SmartCashApplication(context);
        Coin actualSelectedCoin = app.getActualSelectedCoin(context);

        String fiatValue = "";

        if (actualSelectedCoin == null || actualSelectedCoin.getName().equals(context.getString(R.string.default_crypto))) {
            ArrayList<Coin> currentPrice = app.getCurrentPrice(context);
            fiatValue = (app.formatNumberBySelectedCurrencyCode(app.getCurrentValueByRate(this.wallets.get(i).getBalance(), currentPrice.get(0).getValue())));
        } else {
            fiatValue = (app.formatNumberBySelectedCurrencyCode(app.getCurrentValueByRate(this.wallets.get(i).getBalance(), actualSelectedCoin.getValue())));
        }

        walletViewHolder.name.setText(this.wallets.get(i).getDisplayName());
        walletViewHolder.value.setText(app.formatNumberByDefaultCrypto(this.wallets.get(i).getBalance()) + " (" + fiatValue + ")");
        walletViewHolder.address.setText(this.wallets.get(i).getAddress());
        walletViewHolder.btnCopy.setOnClickListener(v -> {
            SmartCashApplication.copyToClipboard(context, this.wallets.get(i).getAddress());
        });

        if (count == 0) {
            @SuppressLint("ResourceType") String color = context.getResources().getString(R.color.btnGreen);
            walletViewHolder.roundIcon.setCardBackgroundColor(Color.parseColor(color));
            Log.e("cont", String.valueOf(count));
            count++;
        } else if (count == 1) {
            @SuppressLint("ResourceType") String color = context.getResources().getString(R.color.btnYellon);
            walletViewHolder.roundIcon.setCardBackgroundColor(Color.parseColor(color));
            Log.e("cont", String.valueOf(count));
            count++;
        } else if (count == 2) {
            @SuppressLint("ResourceType") String color = context.getResources().getString(R.color.btnRed);
            walletViewHolder.roundIcon.setCardBackgroundColor(Color.parseColor(color));
            Log.e("cont", String.valueOf(count));
            count = 0;
        }
    }

    @Override
    public int getItemCount() {
        return wallets.size();
    }

}

File: app/src/main/java/cc/smartcash/smarthub/Adapters/DashboardWalletListAdapter.kt
package cc.smartcash.smarthub.Adapters

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.Color
import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import cc.smartcash.smarthub.Models.Wallet
import cc.smartcash.smarthub.R
import cc.smartcash.smarthub.Utils.SmartCashApplication
import cc.smartcash.smarthub.ViewHolders.DashboardWalletListViewHolder
import java.util.*

class DashboardWalletListAdapter(private val context: Context, private var wallets: ArrayList<Wallet>?) : RecyclerView.Adapter<DashboardWalletListViewHolder>() {

    private var count: Int = 0

    fun setItems(wallets: ArrayList<Wallet>) {
        this.wallets = wallets
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): DashboardWalletListViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.card_wallet_item, viewGroup, false)
        return DashboardWalletListViewHolder(view)
    }

    override fun onBindViewHolder(walletViewHolder: DashboardWalletListViewHolder, i: Int) {

        val app = SmartCashApplication(context)
        val actualSelectedCoin = app.getActualSelectedCoin(context)

        var fiatValue = ""

        fiatValue = if (actualSelectedCoin == null || actualSelectedCoin.name == context.getString(R.string.default_crypto)) {
            val currentPrice = app.getCurrentPrice()
            app.formatNumberBySelectedCurrencyCode(app.getCurrentValueByRate(this.wallets!![i].balance!!, currentPrice!![0].value!!))
        } else {
            app.formatNumberBySelectedCurrencyCode(app.getCurrentValueByRate(this.wallets!![i].balance!!, actualSelectedCoin.value!!))
        }

        walletViewHolder.name.text = this.wallets!![i].displayName
        walletViewHolder.value.text = app.formatNumberByDefaultCrypto(this.wallets!![i].balance!!) + " (" + fiatValue + ")"
        walletViewHolder.address.text = this.wallets!![i].address
        walletViewHolder.btnCopy.setOnClickListener { v -> SmartCashApplication.copyToClipboard(context, this.wallets!![i].address!!) }

        when (count) {
            0 -> {
                @SuppressLint("ResourceType") val color = context.resources.getString(R.color.btnGreen)
                walletViewHolder.roundIcon.setCardBackgroundColor(Color.parseColor(color))
                Log.e("cont", count.toString())
                count++
            }
            1 -> {
                @SuppressLint("ResourceType") val color = context.resources.getString(R.color.btnYellon)
                walletViewHolder.roundIcon.setCardBackgroundColor(Color.parseColor(color))
                Log.e("cont", count.toString())
                count++
            }
            2 -> {
                @SuppressLint("ResourceType") val color = context.resources.getString(R.color.btnRed)
                walletViewHolder.roundIcon.setCardBackgroundColor(Color.parseColor(color))
                Log.e("cont", count.toString())
                count = 0
            }
        }
    }

    override fun getItemCount(): Int {
        return wallets!!.size
    }

}
